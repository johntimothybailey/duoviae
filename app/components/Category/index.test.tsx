import React from 'react'
import { cleanup, render, fireEvent } from '@testing-library/react-native'
import { BootComponent } from '../../../test'
import { Category } from './index'

describe('Category Component', () => {
  describe('Render', () => {
    afterEach(cleanup)
    describe('displays title and icon. Icon by id', () => {
      it('general (fallback)', () => {
        const { toJSON } = render(
          <BootComponent>
            <Category
              item={{
                id: 300,
                name: 'Neuroscience'
              }}
              onPress={() => {}}
            />
          </BootComponent>
        )
        expect(toJSON()).toMatchSnapshot()
      })
      it('Book', () => {
        const { toJSON } = render(
          <BootComponent>
            <Category
              item={{
                id: 10,
                name: 'Comics'
              }}
              onPress={() => {}}
            />
          </BootComponent>
        )
        expect(toJSON()).toMatchSnapshot()
      })
    })
    describe('selection fires event with id', () => {
      const onPress = jest.fn()
      const { queryByText } = render(
        <BootComponent>
          <Category
            item={{
              id: 10,
              name: 'Neuroscience'
            }}
            onPress={onPress}
          />
        </BootComponent>
      )
      fireEvent.press(queryByText('Neuroscience'))
      expect(onPress).toHaveBeenCalledWith({ id: 10, name: 'Neuroscience' })
    })
  })
})
