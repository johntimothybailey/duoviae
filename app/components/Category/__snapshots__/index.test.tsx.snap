// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Component Render displays title and icon. Icon by id Book 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "opacity": 1,
          "paddingHorizontal": 8,
          "paddingVertical": 12,
        }
      }
    >
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={24}
          bbWidth={24}
          fill="#8F9BB3"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "height": 24,
                "marginHorizontal": 8,
                "tintColor": "#8F9BB3",
                "width": 24,
              },
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
        >
          <RNSVGGroup
            fill={4287601587}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGGroup>
              <RNSVGGroup>
                <RNSVGRect
                  height="24"
                  opacity={0}
                  width="24"
                  x={0}
                  y={0}
                />
                <RNSVGPath
                  d="M19 3H7a3 3 0 0 0-3 3v12a3 3 0 0 0 3 3h12a1 1 0 0 0 1-1V4a1 1 0 0 0-1-1zM7 5h11v10H7a3 3 0 0 0-1 .18V6a1 1 0 0 1 1-1zm0 14a1 1 0 0 1 0-2h11v2z"
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 13,
                  "fontWeight": "600",
                  "marginHorizontal": 8,
                },
                Object {
                  "textAlign": "left",
                },
              ],
            ]
          }
        >
          Comics
        </Text>
      </View>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={24}
          bbWidth={24}
          fill="#8F9BB3"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "height": 24,
                "marginHorizontal": 8,
                "tintColor": "#8F9BB3",
                "width": 24,
              },
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
        >
          <RNSVGGroup
            fill={4287601587}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGGroup>
              <RNSVGGroup>
                <RNSVGRect
                  height="24"
                  matrix={
                    Array [
                      6.123233995736766e-17,
                      -1,
                      1,
                      6.123233995736766e-17,
                      0,
                      24,
                    ]
                  }
                  opacity={0}
                  width="24"
                  x={0}
                  y={0}
                />
                <RNSVGPath
                  d="M10.5 17a1 1 0 0 1-.71-.29 1 1 0 0 1 0-1.42L13.1 12 9.92 8.69a1 1 0 0 1 0-1.41 1 1 0 0 1 1.42 0l3.86 4a1 1 0 0 1 0 1.4l-4 4a1 1 0 0 1-.7.32z"
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;

exports[`Category Component Render displays title and icon. Icon by id general (fallback) 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "opacity": 1,
          "paddingHorizontal": 8,
          "paddingVertical": 12,
        }
      }
    >
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={24}
          bbWidth={24}
          fill="#8F9BB3"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "height": 24,
                "marginHorizontal": 8,
                "tintColor": "#8F9BB3",
                "width": 24,
              },
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
        >
          <RNSVGGroup
            fill={4287601587}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGGroup>
              <RNSVGGroup>
                <RNSVGRect
                  height="24"
                  matrix={
                    Array [
                      -1,
                      1.2246467991473532e-16,
                      -1.2246467991473532e-16,
                      -1,
                      24,
                      24,
                    ]
                  }
                  opacity={0}
                  width="24"
                  x={0}
                  y={0}
                />
                <RNSVGPath
                  d="M12 7a5 5 0 0 0-3 9v4a2 2 0 0 0 2 2h2a2 2 0 0 0 2-2v-4a5 5 0 0 0-3-9zm1.5 7.59a1 1 0 0 0-.5.87V20h-2v-4.54a1 1 0 0 0-.5-.87A3 3 0 0 1 9 12a3 3 0 0 1 6 0 3 3 0 0 1-1.5 2.59z"
                />
                <RNSVGPath
                  d="M12 6a1 1 0 0 0 1-1V3a1 1 0 0 0-2 0v2a1 1 0 0 0 1 1z"
                />
                <RNSVGPath
                  d="M21 11h-2a1 1 0 0 0 0 2h2a1 1 0 0 0 0-2z"
                />
                <RNSVGPath
                  d="M5 11H3a1 1 0 0 0 0 2h2a1 1 0 0 0 0-2z"
                />
                <RNSVGPath
                  d="M7.66 6.42L6.22 5a1 1 0 0 0-1.39 1.47l1.44 1.39a1 1 0 0 0 .73.28 1 1 0 0 0 .72-.31 1 1 0 0 0-.06-1.41z"
                />
                <RNSVGPath
                  d="M19.19 5.05a1 1 0 0 0-1.41 0l-1.44 1.37a1 1 0 0 0 0 1.41 1 1 0 0 0 .72.31 1 1 0 0 0 .69-.28l1.44-1.39a1 1 0 0 0 0-1.42z"
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 13,
                  "fontWeight": "600",
                  "marginHorizontal": 8,
                },
                Object {
                  "textAlign": "left",
                },
              ],
            ]
          }
        >
          Neuroscience
        </Text>
      </View>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={24}
          bbWidth={24}
          fill="#8F9BB3"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "height": 24,
                "marginHorizontal": 8,
                "tintColor": "#8F9BB3",
                "width": 24,
              },
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
        >
          <RNSVGGroup
            fill={4287601587}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGGroup>
              <RNSVGGroup>
                <RNSVGRect
                  height="24"
                  matrix={
                    Array [
                      6.123233995736766e-17,
                      -1,
                      1,
                      6.123233995736766e-17,
                      0,
                      24,
                    ]
                  }
                  opacity={0}
                  width="24"
                  x={0}
                  y={0}
                />
                <RNSVGPath
                  d="M10.5 17a1 1 0 0 1-.71-.29 1 1 0 0 1 0-1.42L13.1 12 9.92 8.69a1 1 0 0 1 0-1.41 1 1 0 0 1 1.42 0l3.86 4a1 1 0 0 1 0 1.4l-4 4a1 1 0 0 1-.7.32z"
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
